ser/inference.py

```diff
@@ -0,0 +1,63 @@
+# inference.py
+
+import torch
+from pathlib import Path
+from ser.data import test_dataloader
+from ser.transforms import transforms, normalize
+from ser.params import Params
+import json
+
+def run_infer(run_path: str):
+    label = 6
+
+    # Load saved parameters
+    params_file = Path(run_path) / "params.json"
+    with open(params_file, "r") as f:
+        params_data = json.load(f)
+    params = Params(**params_data)
+
+    # Print experiment summary
+    print("Experiment Name:", params.name)
+    print("Hyperparameters:")
+    print("\tEpochs:", params.epochs)
+    print("\tBatch Size:", params.batch_size)
+    print("\tLearning Rate:", params.learning_rate)
+
```

Comment:
You are mixing what we call high and low-level logic here. Ideally in your cli file- the entry point- we can see the high level logic- load the params, print the params, get the data, get an example of the image with label, get the model then run the inference- the low level details. More abstraction would be good here, and some of this code should stay in the main cli file.